x/pairing/keeper/scores/pairing_slot.go

```diff
@@ -0,0 +1,38 @@
+package scores
+
+import "strings"
+
+// PairingSlot holds the set of requirements for a slot.
+// It also holds the number of identical slots required for the pairing (count)
+type PairingSlot struct {
+	Reqs  map[string]ScoreReq
+	Count uint64
+}
+
+func NewPairingSlot() *PairingSlot {
+	return &PairingSlot{Count: 1}
+}
+
+// Subtract generates a diff slot that contains the reqs that are in the slot receiver but not in the "other" slot
+func (s PairingSlot) Subtract(other *PairingSlot) *PairingSlot {
+	reqsDiff := make(map[string]ScoreReq)
+	for key := range s.Reqs {
+		if _, found := other.Reqs[key]; !found {
+			reqsDiff[key] = s.Reqs[key]
+		}
+	}
+
+	diffSlot := NewPairingSlot()
+	diffSlot.Reqs = reqsDiff
+	return diffSlot
+}
+
+// GetSlotKey generates a unique key of the slot based on its requirements
+func (s PairingSlot) GetSlotKey() string {
+	key := ""
```

Comment:
how about:
```
    keys := make([]string, 0, len(s.Reqs))
    for k := range s.Reqs {
        keys = append(keys, k)
    }
    return strings.Join(keys, "-")
```

Reply:
I changed GroupSlots to be deterministic without relying on a slot key. see https://github.com/lavanet/lava/commit/be6c5597c344f0ad20e94028ffee8928abae3543